src/components/address-output/_macro.spec.js

```diff
@@ -5,94 +5,141 @@ import * as cheerio from 'cheerio';
-
-        const results = await axe($.html());
-        expect(results).toHaveNoViolations();
-    });
+import { EXAMPLE_ADDRESS_OUTPUT_FULL } from './_test_examples';
 
-    it('has additionally provided container style classes', () => {
-        const $ = cheerio.load(
-            renderComponent('address-output', {
-                ...EXAMPLE_ADDRESS_OUTPUT_FULL,
-                classes: 'extra-class another-extra-class',
-            }),
-        );
+describe('FOR: address-output', () => {
+    describe('GIVEN: Params: none', () => {
+        describe('WHEN: no parameters are provided', () => {
+            const $ = cheerio.load(renderComponent('address-output', {}));
 
-        expect($('.ons-address-output').hasClass('extra-class')).toBe(true);
-        expect($('.ons-address-output').hasClass('another-extra-class')).toBe(true);
+            test('THEN: renders no lines', () => {
+                expect($('.ons-address-output__lines *').length).toBe(0);
+            });
+        });
     });
 
-    it('renders no lines when no parameters are provided', () => {
-        const $ = cheerio.load(renderComponent('address-output', EXAMPLE_ADDRESS_OUTPUT_NONE));
-
-        expect($('.ons-address-output__lines *').length).toBe(0);
+    describe('GIVEN: Params: classes', () => {
+        describe('WHEN: classes are provided', () => {
+            const $ = cheerio.load(
+                renderComponent('address-output', {
+                    ...EXAMPLE_ADDRESS_OUTPUT_FULL,
+                    classes: 'extra-class another-extra-class',
+                }),
+            );
+
+            test('THEN: renders with additional classes provided', async () => {
+                expect($('.ons-address-output').hasClass('extra-class')).toBe(true);
+                expect($('.ons-address-output').hasClass('another-extra-class')).toBe(true);
+            });
+        });
     });
 
-    it.each([
-        ['all address lines', EXAMPLE_ADDRESS_OUTPUT_FULL],
-        ['single line', { unit: 'Unit 5' }],
-    ])('renders `unit` with %s', (_, params) => {
-        const $ = cheerio.load(renderComponent('address-output', params));
+    describe('GIVEN: Params: All params', () => {
+        describe('WHEN: All address line params are provided', () => {
+            const $ = cheerio.load(renderComponent('address-output', EXAMPLE_ADDRESS_OUTPUT_FULL));
 
-        expect($('.ons-address-output__unit').text().trim()).toBe('Unit 5');
-    });
+            test('THEN: jest-axe tests pass', async () => {
+                const results = await axe($.html());
+                expect(results).toHaveNoViolations();
+            });
 
-    it.each([
-        ['all address lines', EXAMPLE_ADDRESS_OUTPUT_FULL],
-        ['single line', { organisation: 'Trescos' }],
-    ])('renders `organisation` with %s', (_, params) => {
-        const $ = cheerio.load(renderComponent('address-output', params));
+            test('THEN: renders `unit` line with correct text', () => {
```

Comment:
```suggestion
            test('THEN: renders unit with provided text', () => {
```
These tests would be better if they were worded like this